scripts/ci/determine-tests.mjs

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env node
+import {setOutput} from '@actions/core';
+import {readdirSync, statSync} from 'fs';
+import {EOL} from 'os';
+import {basename, dirname, join, relative} from 'path';
+import {getBaseHeadSHAs, getChangedFiles} from './hasFileChanged.mjs';
+import {listImports, ensureFileExists} from './list-imports.mjs';
+
+class NoRelevantChangesError extends Error {
+  constructor() {
+    super('Change detected elsewhere, running no tests.');
+    this.name = 'NoRelevantChangesError';
+  }
+}
+
+class DependentPackageChangeError extends Error {
+  constructor(file) {
+    super(
+      `Change detected in an atomic dependent Coveo package: ${file}. Running all tests.`
+    );
+    this.name = 'DependentPackageChangeError';
+    this.file = file;
+  }
+}
+
+class PackageJsonChangeError extends Error {
+  constructor(file) {
+    super(
+      `Change detected in a package.json or package-lock.json file: ${file}. Running all tests.`
+    );
+    this.name = 'PackageJsonChangeError';
+    this.file = file;
+  }
+}
```

Comment:
Only real change other than the renaming of the file is the handling of those specific case by catching these errors. They were all handled the same way by sending a empty string resulting in every tests running in a single shard.